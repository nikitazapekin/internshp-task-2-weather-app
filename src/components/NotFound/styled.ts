import { media } from "@styles/breakpoints";
import { flex } from "@styles/mixins";
import styled from "styled-components";

export const Wrapper = styled.section`
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;

  ${flex("center", "center", "column")};
`;

export const Content = styled.section`
  max-width: ${({ theme }) => theme.container}px;
  width: 100%;
  padding: ${({ theme }) => theme.spaces.md}px;

  ${flex("center", "center", "column")};

  row-gap: ${({ theme }) => theme.spaces.md}px;
`;

export const Title = styled.section`
  color: ${({ theme }) => theme.colors.black};
  font-weight: ${({ theme }) => theme.fontWeights.regular};
  font-size: ${({ theme }) => theme.fontSizes.xl};
  text-align: center;
`;

export const Text = styled.section`
  color: ${({ theme }) => theme.colors.black};
  font-weight: ${({ theme }) => theme.fontWeights.regular};
  font-size: ${({ theme }) => theme.fontSizes.xs};
  text-align: center;

  ${media.sm`
    font-size: ${({ theme }) => theme.fontSizes.sm};
  `}
`;

export const Image = styled.img`
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  opacity: 0.3;
  object-fit: cover;
  object-position: center;
  z-index: ${({ theme }) => theme.zIndexes.x};

  ${media.sm`
    font-size: ${({ theme }) => theme.fontSizes.xxs};
  `}
`;
