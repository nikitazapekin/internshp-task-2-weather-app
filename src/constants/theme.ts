export const theme = {
  container: 1187,
  deviceHeight: "100vh",
  colors: {
    black: "#000",
    blue: "#122e65",
    gray: "#D9D9D9",
    white: "#fff",
  },
  fontFamilies: {
    primary: "Inter",
    secondary: "sans-serif",
  },
  fontSizes: {
    xl: "41px",
    lg: "30px",
    md: "27px",
    sm: "22px",
    xs: "20px",
    xxs: "18px",
  },
  fontWeights: {
    regular: 400,
  },
  media: {
    xxl: 1400,
    xl: 1280,
    lg: 1024,
    md: 768,
    xh: 560,
    sm: 480,
  },
  zIndexes: {
    x: -1,
    xs: 0,
    sm: 1,
    md: 2,
  },
  spaces: {
    x: 5,
    xs: 10,
    sm: 15,
    md: 20,
    lg: 30,
    xl: 40,
    xxl: 50,
    g: 60,
    h: 70,
    xh: 80,
    eh: 90,
  },
  swiper: {
    item_gap: 5,
    margin_base: 2.5,
    vw_ratio: 0.5,
    scroll_snap_type: "x mandatory",
    scroll_snap_align: "start",
    flex_basis: "0 0",
    min_width: "0",
    container: 16.6,
  },
};
