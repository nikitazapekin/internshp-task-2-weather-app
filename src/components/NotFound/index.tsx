import { useNavigate } from "react-router-dom";
import NoGeolocationBackground from "@assets/NoGeolocationBackground.webp";
import Button from "@components/Button";
import { UI_CONSTANTS } from "@constants";
import { MAIN_PAGE } from "@constants";

import { Content, Image, Text, Title, Wrapper } from "./styled";

const NotFound = () => {
  const { backToHomepage } = UI_CONSTANTS.buttons;
  const navigate = useNavigate();
  const handleNavigateToHomepage = () => {
    void navigate(MAIN_PAGE);
  };

  return (
    <Wrapper data-testid="not-found-page">
      <Image src={NoGeolocationBackground} />
      <Content>
        <Title>Page Not Found</Title>
        <Text> The page you have reached does not exist</Text>
        <Button text={backToHomepage} handler={handleNavigateToHomepage} />
      </Content>
    </Wrapper>
  );
};

export default NotFound;
