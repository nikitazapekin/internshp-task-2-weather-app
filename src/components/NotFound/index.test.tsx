import { NOT_FOUND_TEST } from "@constants";
import { theme } from "@constants/theme";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import NotFound from ".";

const { DESCRIPTION, IT, CONSTANTS } = NOT_FOUND_TEST;
const {
  RENDERS_TITLE_AND_MESSAGE,
  RENDERS_NAVIGATION_BUTTON,
  RENDERS_BACKGROUND_IMAGE,
  MATCHES_SNAPSHOT,
} = IT;
const { TEXT, IMAGE, BUTTON } = CONSTANTS;

jest.mock("@assets/NoGeolocationBackground.webp", () => "test-image.webp");

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => jest.fn(),
}));

describe(`${DESCRIPTION}`, () => {
  test(`${RENDERS_TITLE_AND_MESSAGE}`, () => {
    render(
      <ThemeProvider theme={theme}>
        <NotFound />
      </ThemeProvider>
    );

    expect(screen.getByText(TEXT.PAGE_TITLE)).toBeInTheDocument();
    expect(screen.getByText(TEXT.PAGE_MESSAGE)).toBeInTheDocument();
  });

  test(`${RENDERS_NAVIGATION_BUTTON}`, () => {
    render(
      <ThemeProvider theme={theme}>
        <NotFound />
      </ThemeProvider>
    );

    const button = screen.getByRole("button");

    expect(button).toHaveTextContent(BUTTON.BACK_TO_HOMEPAGE);
  });

  test(`${RENDERS_BACKGROUND_IMAGE}`, () => {
    render(
      <ThemeProvider theme={theme}>
        <NotFound />
      </ThemeProvider>
    );

    const image = screen.getByRole("img");

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("src", IMAGE.MOCK_SRC);
  });

  test(`${MATCHES_SNAPSHOT}`, () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <NotFound />
      </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
