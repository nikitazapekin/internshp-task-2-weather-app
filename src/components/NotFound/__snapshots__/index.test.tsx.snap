// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NotFound Component matches basic structure snapshot 1`] = `
<section
  class="sc-hvigdm jFOnpX"
  data-testid="not-found-page"
>
  <img
    class="sc-jwTyAe enONoM"
    src="test-image.webp"
  />
  <section
    class="sc-fhHczv gcRCkT"
  >
    <section
      class="sc-ggWZvA dWXBqe"
    >
      Page Not Found
    </section>
    <section
      class="sc-dTvVRJ evOKyb"
    >
       The page you have reached does not exist
    </section>
    <button
      class="sc-bRKDuR bcDZqI"
    >
      Home
    </button>
  </section>
</section>
`;
